experimental/adaptive-tool/README.md

```diff
@@ -12,32 +12,79 @@
 - press `F1` then type `LG live test` to start LG webview test
 
 ## Features
-### Debugging adaptive dialogs
-- Supporting adaptive dialog breaking point
-
-### Debugging language generation
-- Support LG template breaking points
-- Support Expression breaking points
+### Debugging
+[Setting up](#set_up)
+- supporting adaptive dialog breaking point
+- support LG template breaking points
+- support Expression breaking points
 
 ### Syntax highlighting, auto-suggest, auto complete functionality
 #### .lu documents
-- Syntax highlighting
-
+- syntax highlighting
+![lu_syntax_highlighting](./resources/images/lu_syntax_highlighting.png)
+- diagnostic check
+![lu_diagnostic](./resources/images/lu_diagnostic.png)
+- completion 
+![lu_completion](./resources/images/lu_completion.gif)
 #### .lg documents
-- Syntax highlighting
+- syntax highlighting
+![lg_syntax_highlighting](./resources/images/lg_syntax_highlighting.png)
+
+- diagnostic check
+![lg_diagnostic](./resources/images/lg_diagnostic.gif)
+
 - template reference hover
+![template_hover](./resources/images/template_hover.png)
+
 - builtin function hover
-- buildin function suggestion
-- template reference suggestion
+![function_hover](./resources/images/function_hover.png)
+
+- buildin function and template suggestion
+![function_template_suggestion](./resources/images/function_template_suggestion.gif)
+
+- structure property suggestion
+![structure_suggestion](./resources/images/structure_suggestion.gif)
+
 - template definition
-- error check
+![template_definition](./resources/images/template_definition.gif)
+
 
 ### Expansion/ test UI for .lg documents
-- template evaluation
+- template/free text evaluation
+Press `F1` and select `LG live tester` to start LG tester.
+This tool can be used to test specific template or free inline text.
+
+<a name="set_up"></a>
+
+## Setting up and using Visual Studio Code to use the debugger
+### setting up
+To configure Visual Studio Code you need to add a target in your launch.settings file.
+
+You can do that by the **add a configuration** in the debug panel.
+
+There should be 2 configuration templates available:
+
+* **Bot: Launch .NET Core Configuration** - Configuration for building and launching your bot via **dotnet run** and connecting to it
+* **Bot: Attach Configuration** - Configuration for attaching to the debug port of an already running bot (such as IIS express)
+
+### Using
+
+* Open any source file (*.dialog, *.lg) and set breakpoints.
+* Hit F5 to start debugger.
+
+As you interact with the bot your breakpoint should hit and you should be able to inspect memory, call context, etc.
```

Comment:
Add a screen shot showing/ highlighting where to inspect memory, call context etc.